exercises/palindrome-products/palindrome_products_test.go

```diff
@@ -37,10 +38,14 @@ var testData = []struct {
 		""},
 	{4, 10, Product{}, Product{}, "No palindromes"},
 	{10, 4, Product{}, Product{}, "fmin > fmax"},
-	/* bonus curiosities.  (can a negative number be a palindrome?
-	// most say no
-	{-99, -10, Product{}, Product{}, "Negative limits"},
-	// but you can still get non-negative products from negative factors.
+
+	/* Bonus curiosities.  Can a negative number be a palindrome? Most say no
+	{-99, -10,
+		Product{},
+		Product{},
+	"Negative limits"},
```

Comment:
I had a different experience: It compiled fine for me, but the example solution doesn't handle the case.

The first problem with the example code (well, more like a weird thing with this test case, really) that zero is a palindrome, so all pairs whose product is zero was getting appended to the array, which for `0..maxint` is going to be a large number of them (so much that you can easily run out of memory).

The other problem of course is that the example solution doesn't deal with the overflow.

Even if the 0 were changed to a 1, a solution that doesn't detect the overflow would take a long time to run (it does take a long time to make maxint^2 iterations, of course). You can ctrl-C the test, of course, but it can still be puzzling.

I think if this case is kept, it is wise to have a comment warning of the potential problems. I would not mind removing it.